glean-core/ios/GleanTests/Scheduler/MetricsPingSchedulerTests.swift

```diff
@@ -0,0 +1,331 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@testable import Glean
+import OHHTTPStubs
+import XCTest
+
+class MetricsPingSchedulerTests: XCTestCase {
+    var expectation: XCTestExpectation?
+
+    private func setupHttpResponseStub(statusCode: Int32 = 200) {
+        let host = URL(string: Configuration.Constants.defaultTelemetryEndpoint)!.host!
+        stub(condition: isHost(host)) { data in
+            let body = (data as NSURLRequest).ohhttpStubs_HTTPBody()
+            let json = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
+            XCTAssert(json != nil)
+            XCTAssertEqual(json?["ping"] as? String, "test")
+
+            return OHHTTPStubsResponse(
+                jsonObject: [],
+                statusCode: statusCode,
+                headers: ["Content-Type": "application/json"]
+            )
+        }
+    }
+
+    private func clearPingDirectory() {
+        // Get the ping directory
+        let pingDir = HttpPingUploader(configuration: Configuration())
+            .getOrCreatePingDirectory()
+
+        // Clear the directory to ensure we start fresh
+        // Verify all the files were removed, including the bad ones
+        do {
+            let directoryContents = try FileManager.default.contentsOfDirectory(
+                atPath: pingDir.relativePath
+            )
+            for file in directoryContents {
+                try FileManager.default.removeItem(
+                    atPath: pingDir.appendingPathComponent(file).relativePath
+                )
+            }
+        } catch {
+            // Do nothing
+        }
+    }
+
+    override func setUp() {
+        Glean.shared.enableTestingMode()
+    }
+
+    func testIsAfterDueTime() {
+        let mps = MetricsPingScheduler()
+        var fakeNow = DateComponents()
+        var fakeDate = Date()
+
+        // Must report false before the due time on the same calendar day
+        // Shortly before
+        fakeNow.year = 2015
+        fakeNow.month = 6
+        fakeNow.day = 11
+        fakeNow.hour = 3
+        fakeDate = Calendar.current.date(from: fakeNow)!
+        XCTAssertFalse(
+            mps.isAfterDueTime(fakeDate, dueHourOfTheDay: 4),
+            "isAfterDueTime must report false before the due time on the same calendar day"
+        )
+        // The same hour
+        fakeNow.year = 2015
+        fakeNow.month = 6
+        fakeNow.day = 11
+        fakeNow.hour = 4
```

Comment:
I see below we set minute to 1 later on.
Can we explicitly set it to 0 here to make this clear?